validator/sawtooth_validator/consensus/handlers.py

```diff
@@ -144,6 +144,29 @@ def handle_request(self, request, response):
             request.name,
             request.version)
 
+        return HandlerStatus.RETURN_AND_PASS
+
+
+class ConsensusRegisterBlockNewSyncHandler(Handler):
+    def __init__(self, proxy, consensus_notifier):
+        self._proxy = proxy
+        self._consensus_notifier = consensus_notifier
+
+    @property
+    def request_type(self):
+        return validator_pb2.Message.CONSENSUS_REGISTER_REQUEST
+
+    def handle(self, connection_id, message_content):
+        forks = self._proxy.forks()
+
+        if not forks:
+            return None
```

Comment:
What will happen on the consensus side when it's waiting on the future to resolve for this message.  Does it get a response?

Reply:
Shouldn't this return a `HandlerStatus`?

Reply:
It should return HandlerResult